### PR TITLE
Bump spdlog 1.7.0

### LINKED ISSUE #2228: [package] spdlog/1.6.1: embedded fmt conflicts with conan fmt
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **spdlog/1.6.1**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 2019**
  * Conan version: **conan 1.26.0**
  * Python version: **Python 3.7.2**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
* create a conan file with a requirement on fmt/7.0.1 and spdlog/1.6.1
* when building spdlog gets an error because it embeds fmt/6.2.1
* workaround is to use fmt/6.2.1

This one is admittedly a bit weird...it's really a library problem but it can potentially be fixed in the recipe. I guess the most you can hope for is that spdlog pulls in fmt/6.2.1 via a conan requirement, and then you get a nice error message from conan saying there's a conflict rather than trying to extract the real problem from the compilation errors.

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **spdlog/1.7.0**

* Bump spdlog to 1.7.0
* Add fmt 7.1.0
* Restrict maximum version of fmt for spdlog

fixes #2228

/cc @rconde01

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
#2174 duplicate

File: recipes/spdlog/all/conanfile.py [hunk 1]
```diff
@@ -82,6 +84,9 @@ def _disable_werror(self):
         tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "utils.cmake"), "/WX", "")
 
     def build(self):
+        if tools.Version(self.version) < "1.7" and \
+           tools.Version(self.deps_cpp_info["fmt"].version) >= 7:
+            raise ConanInvalidConfiguration("The project spdlog {} requires fmt <7.x".format(self.version))
```
Comment (prince-chrismc) [inline recipes/spdlog/all/conanfile.py:89]: Since there's API/ABI breaks in fmt, should this check be added for the other versions as well?

Comment (uilianries) [reply recipes/spdlog/all/conanfile.py:89]: Yes, if you know the version. Please, copy this valid to your PR, so we can fix #2228

**Comment by gocarlos** [comment]
close this one as #2174 is merged?

**Comment by uilianries** [comment]
@gocarlos Yes!

**Pull request closed without merging**